README.md

```diff
@@ -102,16 +105,71 @@ $ ipfs config --json API.HTTPHeaders.Access-Control-Allow-Origin "[\"http://exam
 
 ### API
 
-> `WIP`
-
-`js-ipfs-api` follows the spec defined by [`interface-ipfs-core`](https://github.com/ipfs/interface-ipfs-core), which concerns the interface to expect from IPFS implementations. This interface is a currently active endeavor - expect it to be complete in the next few weeks (August 2016). You can use it today to consult the methods available.
+> `js-ipfs-api` follows the spec defined by [`interface-ipfs-core`](https://github.com/ipfs/interface-ipfs-core), which concerns the interface to expect from IPFS implementations. This interface is a currently active endeavor - expect it to be complete in the next few weeks (August 2016). You can use it today to consult the methods available.
 
-### Extra API methods
+### Utility functions
 
-Adding to the methods defined by [`interface-ipfs-core`](https://github.com/ipfs/interface-ipfs-core), `js-ipfs-api` exposes a set of extra utility methods.
+Adding to the methods defined by [`interface-ipfs-core`](https://github.com/ipfs/interface-ipfs-core), `js-ipfs-api` exposes a set of extra utility methods. These utility functions are scoped behind the `ipfs.util`.
 
 Complete documentation for these methods is coming with: https://github.com/ipfs/js-ipfs-api/pull/305
 
+
+#### Add files or entire directories from the FileSystem to IPFS
+
+> `ipfs.util.fsAdd(path, option, callback)`
+
+Reads path from disk (if directory add an options `{ recursive: true }` and adds it to IPFS.
+
+```JavaScript
+ipfs.util.fsAdd(<dirpath>, { recursive: true }, (err, result) => {
+  if (err) {
+    throw err
+  }
+  console.log(result)
+})
+```
+
+`files` is an array of objects describing the files that were added, such as:
+
+```
+[{
+  path: 'test-folder',
+  hash: 'QmRNjDeKStKGTQXnJ2NFqeQ9oW23WcpbmvCVrpDHgDg3T6',
+  size: 2278
+},
+// ...
+]
+```
+
+#### Add a file from a URL to IPFS
+
+> `ipfs.util.urlAdd(url, callback)`
+
+```JavaScript
+ipfs.util.urlAdd('http://example.com/', (err, result) => {
+  if (err) {
+    throw err
+  }
+  console.log(result)
+})
+
+```
+
+#### Add a file from a stream to IPFS
+
+> `ipfs.util.streamAdd(stream, callback)`
+
+This is very similar to `ipfs.files.add({path:'', content: stream})`. It is like the reverse of cat
+
+```JavaScript
+ipfs.util.streamAdd(<readable-stream>, (err, result) => {
+  if (err) {
+    throw err
+  }
+  console.log(result)
+})
+```
+
```

Comment:
This might be too much bikeshedding, but I was thinking the naming of the functions. Instead of .fsAdd(), .streamAdd(), urlAdd() how about:
.addFromStream()
.addFromUrl()
.addFromFs()

I think switching the wording would make it a little from consistent and intuitive.